src/iris/orchestration/output_builders.py

```diff
@@ -6,6 +6,43 @@
 from iris.io.dataclasses import ImmutableModel
 
 
+def build_simple_output(call_trace: PipelineCallTraceStorage) -> Dict[str, Any]:
+    """Build the output for the Orb.
+
+    Args:
+        call_trace (PipelineCallTraceStorage): Pipeline call results storage.
+
+    Returns:
+        Dict[str, Any]: {
+                "iris_template": (Optional[IrisTemplate]) the iris template object if the pipeline succeeded,
+                "error": (Optional[Dict]) the error dict if the pipeline returned an error,
+                "metadata": (Dict) the metadata dict,
+        }.
+    """
+    metadata = __get_metadata(call_trace=call_trace)
+    error = __get_error(call_trace=call_trace)
+    iris_template = None
+
+    exception = call_trace.get_error()
+    if exception is None:
+        iris_template = call_trace["encoder"]
+        error = None
+    elif isinstance(exception, Exception):
+        error = {
+            "error_type": type(exception).__name__,
+            "message": str(exception),
+            "traceback": "".join(traceback.format_tb(exception.__traceback__)),
+        }
```

Comment:
@wiktorlazarski This is already dealt with in  `__get_error`, we can simplify this function even more 🙂

Reply:
Nice! I completely missed that. Thanks for the hint. That'll allow us to simplify also `build_orb_output` function.